### PR TITLE
support gitlab target and dynamic secret

### PR DESCRIPTION
**Pull request by eitan-kr**

### CODE REVIEW

File: akeyless/resource_dynamic_secret_gitlab.go [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Dynamic secret name",
+				ForceNew:    true,
+			},
+			"target_name": {
+				Type:        schema.TypeString,
+				Required:    false,
+				Optional:    true,
```
Comment (tuvia-akeyless) [inline akeyless/resource_dynamic_secret_gitlab.go:36]: Everywhere the field **Optional** is true, you can remove the field **Required**
Also in gitlab-target...

File: akeyless/resource_dynamic_secret_gitlab.go [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+				Description: "Access Token TTL",
+				Default:     "60m",
+			},
+			"gitlab_access_token": {
+				Type:        schema.TypeString,
+				Required:    false,
+				Optional:    true,
+				Description: "Gitlab access token",
```
Comment (tuvia-akeyless) [inline akeyless/resource_dynamic_secret_gitlab.go:81]: Add `Sensitive: true`

File: akeyless/resource_dynamic_secret_gitlab.go [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+			},
+			"gitlab_certificate": {
+				Type:        schema.TypeString,
+				Required:    false,
+				Optional:    true,
+				Description: "Gitlab tls certificate (base64 encoded)",
```
Comment (tuvia-akeyless) [inline akeyless/resource_dynamic_secret_gitlab.go:87]: Add `Sensitive: true`

File: akeyless/resource_dynamic_secret_gitlab.go [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+				Required:    false,
+				Optional:    true,
+				Description: "Gitlab base url",
+				Default:     "https://gitlab.com/",
+			},
+			"tags": {
+				Type:        schema.TypeSet,
+				Required:    false,
+				Optional:    true,
+				Description: "Add tags attached to this object. To specify multiple tags use argument multiple times: --tag Tag1 -t Tag2",
```
Comment (tuvia-akeyless) [inline akeyless/resource_dynamic_secret_gitlab.go:100]: please see tags in tests:
```["abc", "def"]```

File: akeyless/resource_producer_test.go [hunk 5]
```diff
@@ -117,6 +118,47 @@ func TestGithubProducerResource(t *testing.T) {
 	tesItemResource(t, config, configUpdate, itemPath)
 }
 
+func TestGitlabProducerResource(t *testing.T) {
+
+	t.Skip("for now the requested values are fictive")
```
Comment (tuvia-akeyless) [inline akeyless/resource_producer_test.go:123]: - Please run this test locally (after success put back the skip)
- Please in update, where there are default values, omit the field empty

File: akeyless/resource_dynamic_secret_gitlab.go [hunk 6]
```diff
@@ -0,0 +1,352 @@
...
+			if res.StatusCode == http.StatusNotFound {
+				// The resource was deleted outside of the current Terraform workspace, so invalidate this resource
+				d.SetId("")
+				return nil
+			}
+			return fmt.Errorf("can't value: %v", string(apiErr.Body()))
+		}
+		return fmt.Errorf("can't get value: %v", err)
+	}
+	if rOut.GitlabAccessType != nil {
```
Comment (tuvia-akeyless) [inline akeyless/resource_dynamic_secret_gitlab.go:200]: care `out.<...>.Metadata`

File: akeyless/resource_gitlab_target.go [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Target name",
+				ForceNew:    true,
+			},
+			"gitlab_access_token": {
+				Type:        schema.TypeString,
+				Required:    false,
+				Optional:    true,
+				Description: "Gitlab access token",
```
Comment (tuvia-akeyless) [inline akeyless/resource_gitlab_target.go:35]: sensitive

File: akeyless/resource_gitlab_target.go [hunk 8]
```diff
@@ -0,0 +1,242 @@
...
+			},
+			"gitlab_certificate": {
+				Type:        schema.TypeString,
+				Required:    false,
+				Optional:    true,
+				Description: "Gitlab tls certificate (base64 encoded)",
```
Comment (tuvia-akeyless) [inline akeyless/resource_gitlab_target.go:41]: sensitive

File: akeyless/resource_gitlab_target.go [hunk 9]
```diff
@@ -0,0 +1,242 @@
...
+	var apiErr akeyless_api.GenericOpenAPIError
+	ctx := context.Background()
+	name := d.Get("name").(string)
+	gitlabAccessToken := d.Get("gitlab_access_token").(string)
+	gitlabCertificate := d.Get("gitlab_certificate").(string)
+	gitlabUrl := d.Get("gitlab_url").(string)
+	description := d.Get("description").(string)
+	key := d.Get("key").(string)
+
+	body := akeyless_api.CreateGitlabTarget{
```
Comment (tuvia-akeyless) [inline akeyless/resource_gitlab_target.go:80]: What's about the new SDK with the sub commands?

File: akeyless/resource_target_test.go [hunk 10]
```diff
@@ -35,6 +35,30 @@ func TestGithubTargetResource(t *testing.T) {
...
+func TestGitlabTargetResource(t *testing.T) {
+	secretName := "gitlab_test"
+	secretPath := testPath("gitlab_target1")
+	config := fmt.Sprintf(`
+		resource "akeyless_target_gitlab" "%v" {
+			name 				= "%v"
+			gitlab_access_token = "aaaaa"
+			gitlab_certificate  = "1234"
+			description 		= "eeeee"
+		}
```
Comment (tuvia-akeyless) [inline akeyless/resource_target_test.go:47]: Try to include as much params as you can.
In the configUpdate, don't include params with default value (`gitlab_url`)

File: version [hunk 11]
```diff
@@ -1,2 +1,2 @@
 # Use Semantic versioning only. Please update the version number before opening a pull request.
-v1.6.3
+v1.7.0
```
Comment (tuvia-akeyless) [inline version:2]: Usually we bump the patch for every new feature. Minor is bumped on massive changes.
(Don't ask me why...)

File: akeyless/resource_gateway_create_k8s_auth_config.go [hunk 12]
```diff
@@ -208,12 +201,6 @@ func resourceK8sAuthConfigRead(d *schema.ResourceData, m interface{}) error {
 			return err
 		}
 	}
-	if rOut.ProtectionKey != nil {
-		err = d.Set("config_encryption_key_name", *rOut.ProtectionKey)
-		if err != nil {
-			return err
-		}
-	}
```
Comment (dan-akeyless) [inline akeyless/resource_gateway_create_k8s_auth_config.go:216]: changes not related to gitlab make sure the behavior is wanted

**Review by tuvia-akeyless — APPROVED** [review]

**Pull request merged**